core_lib/managers/layermanager.cpp

```diff
@@ -261,17 +261,40 @@ bool LayerManager::deleteCurrentLayer()
     return true;
 }
 
+/**
+ * @brief LayerManager::projectLength
+ * @return int: position of the last key frame in the timeline + its length
+ */
 int LayerManager::projectLength()
 {
     int maxFrame = -1;
 
     Object* pObject = editor()->object();
     for ( int i = 0; i < pObject->getLayerCount(); i++ )
     {
-        int frame = pObject->getLayer( i )->getMaxKeyFramePosition();
-        if ( frame > maxFrame )
+        int lastFramePos = pObject->getLayer( i )->getMaxKeyFramePosition();
+        int prevFramePos = pObject->getLayer( i )->getPreviousKeyFramePosition(lastFramePos);
+
+        if (pObject->getLayer(i)->type() == Layer::SOUND) {
+            KeyFrame* keyFrame = pObject->getLayer( i )->getKeyFrameWhichCovers(lastFramePos);
+            KeyFrame* prevKFrame = pObject->getLayer( i )->getKeyFrameWhichCovers(prevFramePos);
+
```

Comment:
Sorry I don't quite get it. What is `prevKFrame` for?

Reply:
prevKFrame is used to access length() and pos() so we can compare the previous keyframe and the last keyframe against each other. This is to prevent cases where you're overlapping soundtracks and the previous frame + its length is longer than the last frame + its length.

Reply:
If so, you might have to do that in all sound clips, not just on the previous one.
example:
```
          +----------------- frame3
    +---------------------------- frame2
+--------------------------------------frame1
```

Reply:
This has been tested against and should work no matter how many clips you use, since it's looping through all layers. 
![test3](https://user-images.githubusercontent.com/1045397/30152115-67ad6448-93b2-11e7-954e-81f93ac5227d.gif)